src/langchain_google_spanner/loader.py

```diff
@@ -0,0 +1,422 @@
+        docs_keys = KeySet(keys=keys)
+        snapshot = database.batch_snapshot()
+        partitions = snapshot.generate_read_batches(
+            table=self.table_name,
+            columns=tuple(self._table_fields),
+            keyset=docs_keys,
+            partition_size_bytes=5000000,
+        )
+        for partition in partitions:
+            keys_to_delete = []
+            for row in snapshot.process_read_batch(partition):
+                # compare whole document
+                if tuple(row) in docs:
+                    keys_to_delete.append([row[0]])
+            with database.batch() as batch:
+                docs_to_delete = KeySet(keys=keys_to_delete)
+                batch.delete(self.table_name, docs_to_delete)
+
+    @staticmethod
+    def init_document_table(
+        instance_id: str,
+        database_id: str,
+        table_name: str,
+        content_column: str = CONTENT_COL_NAME,
+        metadata_columns: List[Column] = [],
+        primary_key: str = "",
+        store_metadata: bool = True,
+        metadata_json_column: str = METADATA_COL_NAME,
+    ):
+        """
+        Create a new table to store docs with a custom schema.
+
+        Args:
+            instance_id: The Spanner instance to load data to.
+            database_id: The Spanner database to load data to.
+            table_name: The table name to load data to.
+            content_column: The name of the content column.
+            metadata_columns: The metadata columns for custom schema.
+            primary_key: The name of the primary key.
+            store_metadata: If true, extra metadata will be stored in the "langchain_metadata" column.
+                            Defaulted to true.
+            metadata_json_column: The name of the special JSON column. Defaulted to use "langchain_metadata".
+        """
+        primary_key = primary_key or content_column
+        client = client_with_user_agent(None, USER_AGENT_LOADER)
+        metadata_json_column = metadata_json_column if store_metadata else ""
+        instance = client.instance(instance_id)
+        if not instance.exists():
+            raise Exception("Instance doesn't exist.")
+        database = instance.database(database_id)
+        if not database.exists():
+            raise Exception("Database doesn't exist.")
+        # create table with custom schema
+        SpannerDocumentSaver.create_table(
+            client,
+            instance_id,
+            database_id,
+            table_name,
+            primary_key,
+            metadata_json_column,
+            content_column,
+            metadata_columns,
+        )
+
+    @staticmethod
+    def create_table(
+        client: Client,
+        instance_id: str,
+        database_id: str,
+        table_name: str,
+        primary_key: str,
+        metadata_json_column: str,
+        content_column: str,
+        metadata_columns: List[Column],
+    ):
+        """Create a new table in Spanner database."""
```

Comment:
Can we add a complete method doc with all the fields explained? as it's a public method.